### PR TITLE
Meshing - Separating the app into core

### PR DESCRIPTION
**Pull request by AlejandroCornejo**

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_elements/fluid_element.h [hunk 1]
```diff
@@ -92,7 +92,11 @@ class FluidElement : public Element
     typedef PointerVectorSet<Dof<double>, IndexedObject> DofsArrayType;
 
     /// Type for shape function values container
-    typedef MatrixRow< Matrix > ShapeFunctionsType;
+    #ifdef KRATOS_USE_AMATRIX
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_elements/fluid_element.h]: this should not be in this PR, I think you need to merge master first

**Review by philbucher — CHANGES_REQUESTED** [review]

File: applications/FluidDynamicsApplication/custom_elements/fluid_element.h (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/FluidDynamicsApplication/custom_elements/fluid_element.h]: idk why the kraken has included this commit :S

Comment (philbucher) [reply applications/FluidDynamicsApplication/custom_elements/fluid_element.h]: > idk why the kraken has included this commit :S

That's what happens when you awaken the Kraken :laughing:

**Review by loumalouomega — APPROVED** [review]
OK now

File: applications/MeshingApplication/meshing_application.h [hunk 2]
```diff
@@ -68,7 +68,7 @@ KRATOS_DEFINE_VARIABLE(double, CONTACT_ANGLE)
 /// Short class definition.
```
Comment (loumalouomega) [inline applications/MeshingApplication/meshing_application.h:68]:
```suggestion
```

**Comment by AlejandroCornejo** [comment]
thx @loumalouomega  and @philbucher

**Review by philbucher — APPROVED** [review]

File: applications/MeshingApplication/meshing_application.h [hunk 3]
```diff
@@ -68,7 +68,7 @@ KRATOS_DEFINE_VARIABLE(double, CONTACT_ANGLE)
...
 /** Detail class definition.
```
Comment (loumalouomega) [inline applications/MeshingApplication/meshing_application.h:69]:
```suggestion
/** 
 * @class KratosMeshingApplication 
 * @brief This is application contains utilities and processes to improve the mesh quality or refine a current mesh
 * @details Additionally the application includes utilities and processes to transfer information to one mesh to a different one or utilities to compute metrics
 * @author Riccardo Rossi
 * @author Pooyan Dadvand
 * @author Jordi Cotela
 * @author Vicente Mataix Ferrandiz
```

File: applications/MeshingApplication/meshing_application.h (hunk 2 above)
Comment (loumalouomega) [inline applications/MeshingApplication/meshing_application.h:68]:
```suggestion
```

**Review by loumalouomega — APPROVED** [review]
I added two suggestions if you want to try this new feature

**Pull request merged**